core/src/test/java/org/elasticsearch/index/query/TemplateQueryIT.java

```diff
@@ -187,16 +188,14 @@ public void testSearchRequestFail() throws Exception {
         searchRequest.indices("_all");
         try {
             String query = "{ \"template\" : { \"query\": {\"match_all\": {}}, \"size\" : \"{{my_size}}\"  } }";
-            BytesReference bytesRef = new BytesArray(query);
-            searchRequest.templateSource(bytesRef);
+            searchRequest.template(parseTemplate(query));
```

Comment:
depending on what we want to test here, does it make sense to take out parsing and just set the template as a java object directly?